### PR TITLE
Remove the `-Wno-stringop-overflow` warning suppression

### PR DESCRIPTION
**Pull request by freak82**
The  `-Wstringop-overflow` warning was caused by single usage of `strncat` function.
The problem was that GCC complained for possibly incorrect usage of `strncat` due to the last argument being equal to the source string length. The last argument of strncat is supposed to be the remaining free space in the destination buffer.
Replaced the usage of `strncat` with the ATS function `ink_strlcat`.
More info about this can be found in [this ATS issue](https://github.com/apache/trafficserver/issues/11343)

### CODE REVIEW

**Comment by freak82** [comment]
I apologize for including the fix about the `memcpy` in `src/iocore/net/UnixUDPNet.cc` in this pull request.
I intended to do separate pull request for this change but ended up pushing the change in the same branch and thus it got into this pull request.
If needed I will close this pull request and do 2 separate ones.

File: src/iocore/net/UnixUDPNet.cc [hunk 1]
```diff
@@ -97,7 +97,7 @@ UDPPacket::new_UDPPacket(struct sockaddr const *to, ink_hrtime when, Ptr<IOBuffe
   p->p.segment_size = segment_size;
 #ifdef HAVE_SO_TXTIME
   if (send_at_hint) {
-    memcpy(&p->p.send_at, &send_at_hint, sizeof(struct timespec));
+    memcpy(&p->p.send_at, send_at_hint, sizeof(struct timespec));
```
Comment (brbzull0) [inline src/iocore/net/UnixUDPNet.cc:100]: Ohh!! this a very good catch.
Must have done some refactor before pushing this and let this slide!! 

Please split this into a different PR just in case.

**Review by brbzull0 — CHANGES_REQUESTED** [review]

File: src/iocore/net/UnixUDPNet.cc (hunk 1 above)
Comment (freak82) [reply src/iocore/net/UnixUDPNet.cc:100]: Ok. Will close this PR and make new ones.
